src/agenda/index.tsx

```diff
@@ -1,61 +1,93 @@
+// @ts-expect-error
+import {extractComponentProps} from '../component-updater.js';
+// @ts-expect-error
 import {parseDate, xdateToData, toMarkingFormat} from '../interface';
+// @ts-expect-error
 import dateutils from '../dateutils';
+// @ts-expect-error
 import {AGENDA_CALENDAR_KNOB} from '../testIDs';
+// @ts-expect-error
 import {VelocityTracker} from '../input';
+import CalendarList, {CalendarListProps} from '../calendar-list';
 import styleConstructor from './style';
-import CalendarList from '../calendar-list';
-import ReservationList from './reservation-list';
+import ReservationList, {ReservactionListProps}  from './reservation-list';
 
 const HEADER_HEIGHT = 104;
 const KNOB_HEIGHT = 24;
 
+export type ReservationItemType = {
+  name: string;
+  height: number;
+  day: XDate;
+}
+
+export type ReservationsType = {
+  [date: string]: ReservationItemType[]
+}
+
+export type AgendaProps = CalendarListProps & ReservactionListProps & {
+  /** agenda container style */
+  style?: ViewStyle;
+  /** the list of items that have to be displayed in agenda. If you want to render item as empty date
+   the value of date key has to be an empty array []. If there exists no value for date key it is
+   considered that the date in question is not yet loaded */
+  items: ReservationsType;
+  /** callback that gets called when items for a certain month should be loaded (month became visible) */
+  loadItemsForMonth?: (data: any) => void;
+  /** callback that fires when the calendar is opened or closed */
+  onCalendarToggled?: (enabled: boolean) => void;
+  /** callback that gets called on day press */
+  onDayPress?: (data: any) => void;
+  /** callback that gets called when day changes while scrolling agenda list */
+  onDaychange?: (data: any) => void; //TODO: Should be renamed 'onDayChange'
+  /** specify how agenda knob should look like */
+  renderKnob?: () => JSX.Element;
+  /** initially selected day */
+  selected: boolean, //TODO: Should be renamed 'selectedDay'
+  /** Hide knob button. Default = false */
+  hideKnob: boolean,
+  /** When `true` and `hideKnob` prop is `false`, the knob will always be visible and the user will be able to drag the knob up and close the calendar. Default = false */
+  showClosingKnob: boolean
+}
+
+type AgendaState = {
+  scrollY: Animated.Value;
+  calendarIsReady: boolean;
+  calendarScrollable: boolean;
+  firstReservationLoad: boolean;
+  selectedDay: XDate;
+  topDay: XDate;
+}
+
+
 /**
  * @description: Agenda component
  * @extends: CalendarList
  * @extendslink: docs/CalendarList
  * @example: https://github.com/wix/react-native-calendars/blob/master/example/src/screens/agenda.js
  * @gif: https://github.com/wix/react-native-calendars/blob/master/demo/agenda.gif
  */
-export default class AgendaView extends Component {
+export default class AgendaView extends Component<AgendaProps, AgendaState> {
   static displayName = 'Agenda';
-
-  static propTypes = {
```

Comment:
I think we need to keep the prop types for the 'extractComponentProps' function...